airflow_tecton/hooks/tecton_hook.py

```diff
@@ -95,26 +96,43 @@ def _make_request(
         verbose: bool = False,
     ) -> Dict[str, Any]:
         full_path = urllib.parse.urljoin(self.base_url, url)
-        if verbose:
-            logging.info(
-                f"Making POST request to {full_path} with body=\n{self._pformat_dict(data)}"
-            )
-        resp = conn.post(full_path, json.dumps(data))
-        try:
-            resp.raise_for_status()
-        except Exception as e:
-            exc = e
-            if "error" in resp.json():
-                raise Exception(f"Tecton error: {resp.json()['error']}")
-            raise e
-        else:
-            exc = None
-        finally:
-            if verbose or exc:
+        num_retries = 3
+
+        for i in range(num_retries):
+            if verbose:
                 logging.info(
-                    f"Response: Code={resp.status_code} Body=\n{self._pformat_dict(resp.json())}"
+                    f"Making POST request to {full_path} with body=\n{self._pformat_dict(data)}"
                 )
-        return resp.json()
+
+            try:
+                resp = conn.post(full_path, json.dumps(data))
+            except requests.exceptions.ConnectionError as re:
+                exc = re
+                if verbose:
+                    logging.info(
+                        f"Recieved ConnectionError for attempt {i}/{num_retries}. Retrying..."
```

Comment:
nit: should be {i+1} since it's 0 indexed?

Reply:
nice catch